File: Arrays/multy.c
#include <stdio.h>

void multiplyPolynomials(int A[], int degA, int B[], int degB, int result[]) {
    for (int i = 0; i <= degA + degB; i++)
        result[i] = 0;  // Initialize result array with 0

    for (int i = 0; i <= degA; i++) {
        for (int j = 0; j <= degB; j++) {
            result[i + j] += A[i] * B[j];
        }
    }
}

void inputPolynomial(int poly[], int degree, char name) {
    printf("Enter %d coefficients for Polynomial %c (from constant to x^%d):\n", degree + 1, name, degree);
    for (int i = 0; i <= degree; i++) {
        printf("Coefficient of x^%d: ", i);
        scanf("%d", &poly[i]);
    }
}

void printPolynomial(int poly[], int degree) {
    for (int i = degree; i >= 0; i--) {
        if (poly[i] != 0) {
            if (i != degree && poly[i] > 0)
                printf(" + ");
            if (i == 0)
                printf("%d", poly[i]);
            else if (i == 1)
                printf("%dx", poly[i]);
            else
                printf("%dx^%d", poly[i], i);
        }
    }
    printf("\n");
}

int main() {
    int A[100], B[100], result[200];
    int degA, degB;

    printf("Enter degree of Polynomial A: ");
    scanf("%d", &degA);
    inputPolynomial(A, degA, 'A');

    printf("Enter degree of Polynomial B: ");
    scanf("%d", &degB);
    inputPolynomial(B, degB, 'B');

    multiplyPolynomials(A, degA, B, degB, result);

    printf("\nPolynomial A: ");
    printPolynomial(A, degA);

    printf("Polynomial B: ");
    printPolynomial(B, degB);

    printf("Product of A and B: ");
    printPolynomial(result, degA + degB);

    return 0;
}

File: Arrays/multy.py
def multiply_polynomials(A, B):
    # Initialize result array with zeros
    result_degree = len(A) + len(B) - 2
    result = [0] * (result_degree + 1)

    # Multiply each term of A with each term of B
    for i in range(len(A)):
        for j in range(len(B)):
            result[i + j] += A[i] * B[j]

    return result


def input_polynomial(name):
    degree = int(input(f"Enter degree of polynomial {name}: "))
    print(f"Enter {degree + 1} coefficients for {name} from constant term to x^{degree}:")
    coeffs = list(map(int, input().split()))
    while len(coeffs) != degree + 1:
        print(f"Error: Please enter exactly {degree + 1} coefficients.")
        coeffs = list(map(int, input().split()))
    return coeffs


def pretty_print(poly):
    terms = []
    for i in range(len(poly)):
        coeff = poly[i]
        if coeff != 0:
            term = str(coeff)
            if i == 1:
                term += "x"
            elif i > 1:
                term += f"x^{i}"
            terms.append(term)
    return " + ".join(terms[::-1]) if terms else "0"


# ---- Main Program ----
print("Polynomial Multiplication Program")
A = input_polynomial("A")
B = input_polynomial("B")

product = multiply_polynomials(A, B)

print("\nPolynomial A:", pretty_print(A))
print("Polynomial B:", pretty_print(B))
print("Product A * B:", pretty_print(product))
